src/wallet/wallet_tx_builder.cpp

```diff
@@ -12,153 +13,170 @@ int GetAnchorHeight(const CChain& chain, uint32_t anchorConfirmations)
     return nextBlockHeight - anchorConfirmations;
 }
 
-PrepareTransactionResult WalletTxBuilder::PrepareTransaction(
+ChangeAddress
+WalletTxBuilder::GetChangeAddress(
+        CWallet& wallet,
+        const ZTXOSelector& selector,
+        SpendableInputs& spendable,
+        const Payments& resolvedPayments,
+        const TransactionStrategy& strategy,
+        bool afterNU5) const
+{
+    // Determine the account we're sending from.
+    auto sendFromAccount = wallet.FindAccountForSelector(selector).value_or(ZCASH_LEGACY_ACCOUNT);
+
+    auto getAllowedChangePools = [&](const std::set<ReceiverType>& receiverTypes) {
+        std::set<OutputPool> result{resolvedPayments.GetRecipientPools()};
+        // We always allow shielded change when not sending from the legacy account.
+        if (sendFromAccount != ZCASH_LEGACY_ACCOUNT) {
+            result.insert(OutputPool::Sapling);
+        }
+        for (ReceiverType rtype : receiverTypes) {
+            switch (rtype) {
+                case ReceiverType::P2PKH:
+                case ReceiverType::P2SH:
+                    // TODO: This is the correct policy, but it’s a breaking change from
+                    //       previous behavior, so enable it separately. (#6409)
+                    // if (strategy.AllowRevealedRecipients()) {
+                    if (!spendable.utxos.empty() || strategy.AllowRevealedRecipients()) {
```

Comment:
This reverses the change that was merged in #6409, so that should be re-proposed later? Or reproduced here?

Reply:
Oh! I must have lost that in a rebase. Yikes. I’ll restore that. Thanks for catching it!